presto-common/src/main/java/com/facebook/presto/common/block/RowBlock.java

```diff
@@ -0,0 +1,268 @@
+            for (int position = 0; position < fieldBlockOffsets.length; position++) {
+                fieldBlockOffsets[position] = position;
+            }
+        }
+        else {
+            // Check for nulls when computing field block offsets
+            int currentOffset = 0;
+            for (int position = 0; position < positionCount; position++) {
+                fieldBlockOffsets[position] = currentOffset;
+                currentOffset += (rowIsNull[position] ? 0 : 1);
+            }
+            // fieldBlockOffsets is positionCount + 1 in length
+            fieldBlockOffsets[positionCount] = currentOffset;
+            if (currentOffset == positionCount) {
+                // No nulls encountered, discard the null mask
+                rowIsNull = null;
+            }
+        }
+
+        validateConstructorArguments(0, positionCount, rowIsNull, fieldBlockOffsets, fieldBlocks);
+        return new RowBlock(0, positionCount, rowIsNull, fieldBlockOffsets, fieldBlocks);
+    }
+
+    /**
+     * Create a row block directly without per element validations.
+     */
+    static RowBlock createRowBlockInternal(int startOffset, int positionCount, @Nullable boolean[] rowIsNull, int[] fieldBlockOffsets, Block[] fieldBlocks)
+    {
+        validateConstructorArguments(startOffset, positionCount, rowIsNull, fieldBlockOffsets, fieldBlocks);
+        return new RowBlock(startOffset, positionCount, rowIsNull, fieldBlockOffsets, fieldBlocks);
+    }
+
+    private static void validateConstructorArguments(int startOffset, int positionCount, @Nullable boolean[] rowIsNull, int[] fieldBlockOffsets, Block[] fieldBlocks)
+    {
+        if (startOffset < 0) {
+            throw new IllegalArgumentException("arrayOffset is negative");
+        }
+
+        if (positionCount < 0) {
+            throw new IllegalArgumentException("positionCount is negative");
+        }
+
+        if (rowIsNull != null && rowIsNull.length - startOffset < positionCount) {
+            throw new IllegalArgumentException("rowIsNull length is less than positionCount");
+        }
+
+        requireNonNull(fieldBlockOffsets, "fieldBlockOffsets is null");
+        if (fieldBlockOffsets.length - startOffset < positionCount + 1) {
+            throw new IllegalArgumentException("fieldBlockOffsets length is less than positionCount");
+        }
+
+        requireNonNull(fieldBlocks, "fieldBlocks is null");
+
+        if (fieldBlocks.length <= 0) {
+            throw new IllegalArgumentException("Number of fields in RowBlock must be positive");
+        }
+
+        int firstFieldBlockPositionCount = fieldBlocks[0].getPositionCount();
+        for (int i = 1; i < fieldBlocks.length; i++) {
+            if (firstFieldBlockPositionCount != fieldBlocks[i].getPositionCount()) {
+                throw new IllegalArgumentException(format("length of field blocks differ: field 0: %s, block %s: %s", firstFieldBlockPositionCount, i, fieldBlocks[i].getPositionCount()));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: firstFieldBlockPositionCount (declared at line 111), &#39;getPositionCount&#39; expression, i (declared at line 112). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 114. Consider formatting this data with &#39;%d&#39; instead.